### PR TITLE
feat: added group management interfaces

### PR DESCRIPTION
**Pull request by bhudnell**
Issue #, if available:

Description of changes:
- Added `GroupManagementPlugin` interface
- Added `wbcGroupManagementPlugin` method stubs
- merged from develop

Checklist:

* [ ] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: workbench-core/authorization/src/dynamicAuthorization/groupManagementPlugin.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  removeUserFromGroup(request: RemoveUserFromGroupRequest): Promise<RemoveUserFromGroupResponse>;
+
+  /**
+   * Set the status of the group
+   *
+   * @param request - {@link SetGroupStatusRequest}
+   *
+   * @returns a {@link SetGroupStatusResponse}
+   */
+  setGroupStatus(request: SetGroupStatusRequest): Promise<SetGroupStatusResponse>;
```
Comment (htztomic) [inline workbench-core/authorization/src/dynamicAuthorization/groupManagementPlugin.ts:101]: will need a getGroupStatus method

**Comment by htztomic** [comment]
I assume this merges the changes from develop as well

File: workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationInputs/setGroupStatus.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ */
+export interface SetGroupStatusRequest {
+  /**
+   * Group id associated to the group the status is being set on
+   */
+  groupId: string;
+  /**
+   * Status to set
+   */
+  status: string;
```
Comment (htztomic) [inline workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationInputs/setGroupStatus.ts:12]: Make this into a type: status should be only a few options

File: workbench-core/authorization/src/dynamicAuthorization/groupStatus.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Enum of the status of a group
+ */
+export enum GroupStatus {
```
Comment (htztomic) [inline workbench-core/authorization/src/dynamicAuthorization/groupStatus.ts:4]: Doesn't need to be a enum, use types instead.

**Review by FernandoAranda — DISMISSED** [review]

File: workbench-core/authorization/src/dynamicAuthorization/groupStatus.ts (hunk 3 above)
Comment (bhudnell) [reply workbench-core/authorization/src/dynamicAuthorization/groupStatus.ts]: changed.

**Review by htztomic — APPROVED** [review]

**Review by FernandoAranda — APPROVED** [review]

**Pull request merged**